cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/index.tsx

```diff
@@ -0,0 +1,203 @@
+      width: '100%',
+      textAlign: 'left',
+      alignSelf: 'center',
+    },
+    requestPath: {
+      width: `${LEFT_PANEL_WIDTH / 1.5}px`,
+      wordWrap: 'break-word',
+      textAlign: 'left',
+      textTransform: 'lowercase',
+      fontSize: '10px',
+      lineHeight: '1.3',
+    },
+    getMethod: {
+      color: theme.palette.green[50],
+    },
+    postMethod: {
+      color: theme.palette.orange[50],
+    },
+    putMethod: {
+      color: theme.palette.yellow[50],
+    },
+    deleteMethod: {
+      color: theme.palette.red[50],
+    },
+  };
+};
+
+enum RequestMethod {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  DELETE = 'DELETE',
+}
+
+interface IRequestHistory {
+  method: RequestMethod;
+  path: string;
+  body: string;
+  headers: {
+    pairs: [
+      {
+        key: string;
+        value: string;
+        uniqueId: string;
+      }
+    ];
+  };
+  response: string;
+  statusCode: number;
+}
+
+interface IRequestHistoryTabProps extends WithStyles<typeof styles> {
+  incomingRequest: boolean;
+  onRequestClick: (request: IRequestHistory) => void;
+  resetIncomingRequest: () => void;
+}
+
+const mapStateToProps = (state) => {
+  return {
+    incomingRequest: state.http.incomingRequest,
+  };
+};
+
+const mapDispatch = (dispatch) => {
+  return {
+    onRequestClick: (request: IRequestHistory) => {
+      dispatch({
+        type: HttpExecutorActions.setRequestHistoryView,
+        payload: request,
+      });
+    },
+    resetIncomingRequest: () => {
+      dispatch({
+        type: HttpExecutorActions.notifyIncomingRequest,
+        payload: {
+          incomingRequest: false,
+        },
+      });
+    },
+  };
+};
+
+const RequestHistoryTabView: React.FC<IRequestHistoryTabProps> = ({
+  classes,
+  incomingRequest,
+  onRequestClick,
+  resetIncomingRequest,
+}) => {
+  const [requestLog, setRequestLog] = React.useState(List<IRequestHistory>([]));
+
+  // Query through localstorage and popluate RequestHistoryTab
+  React.useEffect(() => {
+    let newRequestLog = List<IRequestHistory>([]);
+    Object.keys(localStorage).forEach((key) => {
+      if (key.startsWith('RequestHistory')) {
+        const newRequest = JSON.parse(localStorage.getItem(key));
+        newRequestLog = newRequestLog.push(newRequest);
+      }
+    });
+    setRequestLog(newRequestLog);
+  }, []);
+
+  // When new request history is incoming, update RequestHistoryTab
+  React.useEffect(() => {
+    if (!incomingRequest) {
+      return;
+    }
+    const timestamp = `RequestHistory ${new Date().toLocaleString()}`;
+    const newRequest = HttpExecutorStore.getState().http;
+
+    // Store new request history in local storage
+    localStorage.setItem(timestamp, JSON.stringify(newRequest));
+
+    // Update the component view in real-time, since we cannot listen to local storage's change
+    const newRequestLog = requestLog.push(newRequest);
+    setRequestLog(newRequestLog);
+
+    resetIncomingRequest();
+  }, [incomingRequest]);
```

Comment:
Since there's already a Redux store, these useEffect hooks should be unnecessary. 

The saving of the request to the localStorage can be done in the ActionCreator. The state of the history should also be from the store.